lib/app/app.rb

```diff
@@ -84,51 +85,92 @@ class ConnectorsWebApp < Sinatra::Base
   end
 
   post '/status' do
-    source_status = @connector.source_status(body_params)
+    connector = settings.connector_class.new
+
+    source_status = connector.source_status(body_params)
     json(
-      :extractor => { :name => @connector.name },
+      :extractor => { :name => connector.name },
       :contentProvider => source_status
     )
   end
 
+  post '/start_sync' do
+    job = settings.job_store.create_job
```

Comment:
Should we reject a type if there's the same type of job for the same content source is running?

Reply:
We don't currently have a concept of job type, so it's hard to do - connector does not distinguish incremental job from full sync job (we can do it by checking `:modified_since` and `:cursors` when starting job.

There is another point to that - we don't have a way to cancel/restart job. So if in Enterprise Search UI or API you restart jobs, then the jobs in the connector will keep their execution (unfortunately) and won't be interrupted (they need to be recycled after completion or hanging). Therefore for now it's not safe to assume there will be single running job of each type on connector.

Reply:
I think it's yet technically not possible as there's a functionality of job cancellation in Enterprise Search that is unable to communicate the fact that job was cancelled to the connector package, and introducing this logic will be a major refactoring.

Since it's too much work it's not something that we'll probably be aiming at currently, therefore I think we can avoid introducing this logic for now (which is an important logic to have in future imo).